aten/src/ATen/core/stack.h

```diff
@@ -9,7 +10,38 @@ namespace jit {
 
 using c10::IValue;
 using Stack = std::vector<IValue>;
-using Operation = std::function<void(Stack*)>;
+
+class Operation {
+ public:
+  template <typename F,
```

Comment:
Instead of switching over to Stack&, we can just add another constructor for Operation, such that it can take both Stack* and Stack& callback. For Stack* constructor we can make it explicit that it's deprecated, which shouldn't be a hard break for users.

Note that we wrap another level of lambda for Stack* callback to make the implementation uniform, which may have performance overhead but we can take it as a forcing factor for users to switch to Stack& callback. (or it can be inlined, I don't know).